core/tests/integration/pass/core/eq.ncl

```diff
@@ -77,5 +77,10 @@ let {check, ..} = import "../lib/assert.ncl" in
         [[1,2,3,4], [1,2,3,4], [1,2,3,4], [1,2,3,4]]
       == false
   ),
+
+  # ADTs
+  'Left..(1+1) == 'Left..(2),
```

Comment:
```suggestion
  'Left..(1+1) == 'Left..(2),
  'Left..(1) != 'Left(2),
```

(just because the case of equal discriminants but unequal payloads isn't covered yet)